### PR TITLE
Document `name` and `id`

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Part of #457

### CODE REVIEW

**Review by DavidGOrtega — APPROVED** [review]
👍 lgtm but I would wait for @casperdcl here

**Comment by 0x2b3bfa0** [comment]
Given that this pull request is a draft, we can wait a bit more. Thanks for the Time Travel Review! 😄

File: docs/resources/task.md [hunk 1]
```diff
@@ -57,18 +57,23 @@ resource "iterative_task" "example" {
 
 -> **Note:** `output` is relative to `workdir`, so `storage { workdir = "foo", output = "bar" }` means "upload `./foo/`, change working directory to the uploaded folder, run `script`, and download `bar` (i.e. `./foo/bar`)".
 
+~> **Warning:** Setting `name` to `Hello, World!` will always produce the `tpi-hello-world-5kz6ldls-57wo7rsp` deterministic identifier. Using the `name` argument is discouraged, and it may be removed or have a different behavior in future versions.
+
 ## Attribute Reference
 
 In addition to all arguments above, the following attributes are exported:
 
+- `id` - Task identifier.
```
Comment (casperdcl) [inline docs/resources/task.md:66]:
```suggestion
- `id` - Task identifier, `tpi-{name}-{random_hash_1}-{random_hash_2}`. Either the full `{id}` or (if too long), the shorter `{random_hash_1}{random_hash_2}` is used as the name for all cloud resources.
```

File: docs/resources/task.md [hunk 2]
```diff
@@ -57,18 +57,23 @@ resource "iterative_task" "example" {
...
 - `ssh_private_key` - Used to access the created machines.
 - `addresses` - IP addresses of the currently active machines.
 - `status` - Status of the machine orchestrator.
 - `events` - List of events for the machine orchestrator.
 - `logs` - List with task logs; one for each machine.
 
-~> **Warning:** Status and events don't produce a stable output between cloud providers and are intended for human consumption only.
+~> **Warning:** `events` doesn't produce a stable output between cloud providers and are intended for human consumption only.
+
+-> **Note:** `id` is always in the form `tpi-{name}–{1}-{2}`, where `{name}`, `{1}` and `{2}` are randomly generated identifiers. All the cloud resources are named `id` when legth limits allow it, or `{1}{2}` in cases where there is a more strict length limit.
```
Comment (casperdcl) [inline docs/resources/task.md:76]:
```suggestion
```

**Review by casperdcl — CHANGES_REQUESTED** [review]

File: docs/resources/task.md [hunk 3]
```diff
@@ -57,18 +57,23 @@ resource "iterative_task" "example" {
...
+- `id` - Task identifier.
 - `ssh_public_key` - Used to access the created machines.
 - `ssh_private_key` - Used to access the created machines.
 - `addresses` - IP addresses of the currently active machines.
 - `status` - Status of the machine orchestrator.
 - `events` - List of events for the machine orchestrator.
 - `logs` - List with task logs; one for each machine.
 
-~> **Warning:** Status and events don't produce a stable output between cloud providers and are intended for human consumption only.
+~> **Warning:** `events` doesn't produce a stable output between cloud providers and are intended for human consumption only.
```
[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply docs/resources/task.md:74]: Yes, `status` is stable since #322

[the remark this answers is not in the document]
Comment (casperdcl) [reply docs/resources/task.md:71]:
```suggestion
~> **Warning:** `events` have different formats across cloud providers and cannot be relied on for programmatic consumption/automation.
```

**Review by casperdcl — APPROVED** [review]

**Pull request merged**